internal/controllers/servicebinding_controller.go

```diff
@@ -50,17 +60,123 @@ type ServiceBindingReconciler struct {
 // For more details, check Reconcile and its Result here:
 // - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.8.3/pkg/reconcile
 func (r *ServiceBindingReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
-	_ = log.FromContext(ctx)
+	log := log.FromContext(ctx)
+
+	servicebinding := &servicebindingv1beta1.ServiceBinding{}
+
+	if err := r.Get(ctx, req.NamespacedName, servicebinding); err != nil {
+		if errors.IsNotFound(err) {
+			// We got a request for reconciliation, but no object exists.  Do
+			// nothing and don't requeue.
+			log.Info("Service Binding not found; ignoring.", "name", req.NamespacedName, "err", err)
+			return ctrl.Result{}, nil
+		}
+		log.Error(err, "Unable to retrieve service binding", "name", req.NamespacedName, "err", err)
+		return ctrl.Result{}, nil
+	}
+	// log.Info("Retrieved Service Binding", "service binding", servicebinding)
+
+	resolver := resolver.New(r.Client)
+	workloadData := servicebinding.Spec.Workload
+	workloadRef := v1.ObjectReference{
+		Name:       workloadData.Name,
+		Namespace:  servicebinding.Namespace,
+		Kind:       workloadData.Kind,
+		APIVersion: workloadData.APIVersion,
+	}
+
+	log.Info("Attempting to retrieve workload", "workloadRef", workloadRef)
+	workload, err := resolver.LookupWorkload(ctx, workloadRef)
+	if err != nil {
+		updateStatus(servicebinding, err)
+		r.Update(ctx, servicebinding, &client.UpdateOptions{})
+
+		log.Error(err,
+			"Unable to retrieve workload",
+			"workload", workloadRef,
+			"service binding", servicebinding)
+		return ctrl.Result{Requeue: true}, nil
+	}
+
+	projector := projector.New(resolver)
+	if servicebinding.DeletionTimestamp.IsZero() {
+		err = projector.Project(ctx, servicebinding, workload)
+		if err != nil {
+			log.Error(
+				err,
+				"Failed to project bindings into workload",
+				"workload", workload,
+				"service binding", servicebinding,
+				"err", err)
```

Comment:
err is already captured
```suggestion
				"service binding", servicebinding)
```